android/app/src/main/java/com/reactnativenavigation/modal/RnnModal.java

```diff
@@ -67,10 +75,17 @@ public void onDisplayed() {
 
     public void push(Screen screen) {
         mScreenStack.push(screen);
+        updateToolbar(screen, mToolBar);
     }
 
     public Screen pop() {
-        return mScreenStack.pop();
+        if (mScreenStack.getStackSize() > 1) {
+            return mScreenStack.pop();
+        } else {
+            ModalController.getInstance().remove();
+            super.onBackPressed();
+            return null;
+        }
```

Comment:
When pop was called on the only remaining screen on a modal it didn't appear to be clearing the modal out properly.